src/linter/block.go

```diff
@@ -960,6 +960,10 @@ func (b *BlockWalker) handleMethodCall(e *expr.MethodCall) bool {
 		magic = haveMagicMethod(typ, `__call`)
 		return foundMethod || magic
 	})
+	// We permit abstract classes to call unimplemented interface methods.
+	if !isAbstractClass(b.r.currentClassNode) && implClass == "" {
```

Comment:
I would change the order of the arguments, e.g. check for implClass == "" first because determining whether or not the class is abstract can be much slower. Also I think you need to only check calls to "$this->" and "self::" (maybe "parent::" too) and not everything because you can call different objects from inside an abstract class and it should still be an error if such methods don't exist. In general you need to check that the class of the _object_ that is being called is abstract and not the current class.

Reply:
Good catch. I tried to fix that, PTAL.